### DISCUSSION TITLE
Creating verifiable credentials returns an error

### DISCUSSION THREAD

**Discussion by marytellezm** [decentralized-identity/veramo#900] [Q&A]
When I try to create a VC it ​​returns the error:
```
..\veramo\node_modules\@veramo\kms-local\build\key-management-system.js:129
                throw new Error(`key_not_found: No key entry found for kid=${keyRef.kid}`);
                      ^

Error: key_not_found: No key entry found for kid=045910558b1a822e7959e86fd8d98c2282925e2edd49afdb3fec251bf5006cd6637a6d458e40f883da0d36222b61c84afdafa6d8d6bb2d2716415b428d99a3a6c7   
    at KeyManagementSystem.<anonymous> (..veramo\node_modules\@veramo\kms-local\build\key-management-system.js:129:23)
    at Generator.throw (<anonymous>)
    at rejected (..veramo\node_modules\@veramo\kms-local\build\key-management-system.js:25:65)
```

This is my code:
 ```
await skills.forEach(async item => {
      const credential = await agent.createVerifiableCredential({
        credential: {
          issuer: {
            id: did,
            name: 'mydomian.com'
           },
          '@context': ['https://www.w3.org/2018/credentials/v1'],
          type: ['VerifiableCredential'],
          issuanceDate: new Date().toISOString(),
          credentialSubject: {
            id: did,
            skill: item.name,
            level: item.value
          },
        },
        proofFormat: 'jwt',
        save: true,
      });
      console.log('Credential created: ', credential);
    });
```
Please help

**Comment by mirceanis** [comment]
There isn't anything wrong with the code you shared, but the error indicates that the issuer DID is referring to a key that is no longer available to the agent.

Did this code ever work?
Has the agent database changed in any way?

**Reply by marytellezm** [reply]
Hi, thit's weird,  my did is in the database:
`yarn id:list`
```
Identifier {
    did: 'did:ethr:rinkeby:0x035910558b1a822e7959e86fd8d98c2282925e2edd49afdb3fec251bf5006cd663',
    provider: 'did:ethr:rinkeby',
    alias: 'myalias',
    controllerKeyId: '045910558b1a822e7959e86fd8d98c2282925e2edd49afdb3fec251bf5006cd6637a6d458e40f883da0d36222b61c84afdafa6d8d6bb2d2716415b428d99a3a6c7',
    keys: [ [Key] ],
    services: []
  },
```

**Reply by mirceanis** [reply]
This seems like an issue with the keys associated with that DID.
Keys can be encrypted in the local database so if the encryption key has changed recently this may be a symptom.

Has anything changed in your agent configuration?
Has this code ever worked before?
Also, what version of Veramo are you using?
Did you recently upgrade to a new version but still using the older database?

**Reply by marytellezm** [reply]
Hi, i created the credential before create the web:did, when i go back this work, 
My agent is the same
I am using veramo 3.0.0
I am using a new database
please help.

**Reply by mirceanis** [reply]
I'm trying to help, but please help me understand the context.
If it is a new database, can you use a new DID too?
Does this happen with every DID?

Can you share a small project where this issue is visible?

**Reply by marytellezm** [reply]
Hi, I changed my create method resolve for did:web and use my domain like alias and it works, the problem is when I create the web:did in the console with `veramo did create`, maybe this is a problem with my agent. Thank you!!!

**Comment by mirceanis** [comment]
I guess (but it is hard to know for sure without more details) that you are using the same database for 2 agents, one controlled from the console and one created by your app.

In that case, please make sure to use the same encryption key for the PrivateKeyStore in your app as the one found in agent.yml


It is impossible to help without knowing what is going on with your setup. If you are still stuck, **please, please, please** help us isolate the problem by providing all the details you can think of (except private keys).

The ideal situation is where you create a small project that  mirrors your setup and shows us the problem.

**Reply by marytellezm** [reply]
hi, this is my setup.ts

```
const DATABASE_FILE = 'database.sqlite';

const INFURA_PROJECT_ID = '<your PROJECT_ID here>';
const KMS_SECRET_KEY = <your KMS_SECRET_KEY here>';

const dbConnection = createConnection({
  type: 'sqlite',
  database: DATABASE_FILE,
  synchronize: false,
  migrations,
  migrationsRun: true,
  logging: ['error', 'info', 'warn'],
  entities: Entities,
});

export const agent = createAgent<IDIDManager & ICredentialIssuer & IMessageHandler & ISelectiveDisclosure & IKeyManager & IDataStore & IDataStoreORM & IResolver>({
  plugins: [
    new KeyManager({
      store: new KeyStore(dbConnection),
      kms: {
        local: new KeyManagementSystem(new PrivateKeyStore(dbConnection, new SecretBox(KMS_SECRET_KEY))),
      },
    }),
    new DIDManager({
      store: new DIDStore(dbConnection),
      defaultProvider: 'did:ethr:rinkeby',
      providers: {
        'did:ethr:rinkeby': new EthrDIDProvider({
          defaultKms: 'local',
          network: 'rinkeby',
          rpcUrl: 'https://rinkeby.infura.io/v3/' + INFURA_PROJECT_ID,
        }),
        'did:web': new WebDIDProvider({
          defaultKms: 'local',
        }),
      },
    }),
    new DIDResolverPlugin({
      resolver: new Resolver({
        ...ethrDidResolver({ infuraProjectId: INFURA_PROJECT_ID }),
        ...webDidResolver(),
      }),
    }),
    new DataStore(dbConnection),
    new DataStoreORM(dbConnection),
    new MessageHandler({
      messageHandlers: [
        new DIDCommMessageHandler(),
        new JwtMessageHandler(),
        new W3cMessageHandler(),
        new SdrMessageHandler(),
      ],
    }),
    new DIDComm(),
    new CredentialIssuer(),
    new SelectiveDisclosure(),
  ],
});
```
Thank you

**Reply by mirceanis** [reply]
Thanks, and are you using the same database to call veramo from the console? You have not confirmed this is the case yet and this is an important question.


If you are calling veramo commands from the console in the same folder as your app, then it is very likely that you are sharing the database.

If so, you have to make sure that the `KMS_SECRET_KEY` from your setup code that you shared above matches the one in `agent.yml`

Can you check that please?

---

Another situation that might lead to this kind of problems is when you use the same database file with different veramo versions. For example, if your app is using Veramo 3.0, but your console is using Veramo 1.2

---

Generally veramo is not prepared to handle this kind of database sharing, at least not yet. So the solution is to avoid database sharing. If you really have to share the database, then make sure the KMS_SECRET_KEY is the same and the veramo versions are the same.
